### PR TITLE
add reset ,reboot ,enable and disable vnc

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 1]
```diff
@@ -180,3 +195,82 @@ def test002_shutdown_and_start_vm(self):
...
+        
+        self.log('Enable vnc_port for [vm1], should succeed.')
+        temp_actions = {'vm': {'actions': ['enable_vnc'], 'service': self.vm1_name}}
+        res = self.create_vm(vms=self.vms, temp_actions=temp_actions)
+        self.assertEqual(type(res), type(dict()))
+        self.wait_for_service_action_status(self.vm1_name, res[self.vm1_name]['enable_vnc'])        
+
+        self.log("Check that vnc_port has been enabled successfully.")
+        vm_vnc_url = '{}:{}'.format(self.zos_redisaddr, self.vm_vnc_port)
+        result = self.check_vnc(vm_vnc_url)
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: Should this be url or ip?

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 2]
```diff
@@ -180,3 +195,82 @@ def test002_shutdown_and_start_vm(self):
...
+        self.assertNotIn("timeout caused connection failure", result)
+
+        self.log("Disable vnc_port for [vm1], should succeed.")
+        temp_actions = {'vm': {'actions': ['disable_vnc'], 'service': self.vm1_name}}
+        res = self.create_vm(vms=self.vms, temp_actions=temp_actions)
+        self.assertEqual(type(res), type(dict()))
+        self.wait_for_service_action_status(self.vm1_name, res[self.vm1_name]['disable_vnc'])        
+
+        self.log("Check that vnc_port has been disabled successfully.")
+        vm_vnc_url = '{}:{}'.format(self.zos_redisaddr, self.vm_vnc_port)
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: this line is repeated with no need?

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 3]
```diff
@@ -180,3 +195,82 @@ def test002_shutdown_and_start_vm(self):
         self.assertEqual(vm1[0]['state'], "running")
 
         self.log('%s ENDED' % self._testID)
+
+    def test003_enable_and_disable_vm(self, action_type):
+        """ ZRT-ZOS-007
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: this id is not matching with name in the ids sheet: https://docs.google.com/spreadsheets/d/13xMDt_aNLshO5s0Htbc2FQo98m7I0CDMusJBy2gDN6s/edit#gid=86315193

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 4]
```diff
@@ -180,3 +195,82 @@ def test002_shutdown_and_start_vm(self):
...
+        self.assertEqual(type(res), type(dict()))
+        self.wait_for_service_action_status(self.vm1_name, res[self.vm1_name]['disable_vnc'])        
+
+        self.log("Check that vnc_port has been disabled successfully.")
+        vm_vnc_url = '{}:{}'.format(self.zos_redisaddr, self.vm_vnc_port)
+        result = self.check_vnc(vm_vnc_url)       
+        self.assertIn("timeout caused connection failure", result)
+
+    @parameterized.expand(["reset", "reboot"])
+    def test004_reset_and_reboot_vm(self, action_type):
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: This id with the test name is in the sheet: https://docs.google.com/spreadsheets/d/13xMDt_aNLshO5s0Htbc2FQo98m7I0CDMusJBy2gDN6s/edit#gid=86315193 .. Also please make sure that the names are matching the ids for all vms test cases.

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 5]
```diff
@@ -180,3 +195,82 @@ def test002_shutdown_and_start_vm(self):
...
+        """ ZRT-ZOS-008
+        *Test case for testing reset vm*
+
+        **Test Scenario:**
+
+        #. Create a vm[vm1]  on node, should succeed.
+        #. Enable vnc_port for [vm1], should succeed.
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: doesn't make sense to enable the vm in another test case

Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: why are the order of the steps not like that of the test case?

File: tests/integration_tests/testsuite/a_basic/vms_tests.py [hunk 6]
```diff
@@ -180,3 +195,82 @@ def test002_shutdown_and_start_vm(self):
...
+        time.sleep(10)
+
+        self.log('Enable vnc_port for [vm1], should succeed.')
+        temp_actions = {'vm': {'actions': ['enable_vnc'], 'service': self.vm1_name}}
+        res = self.create_vm(vms=self.vms, temp_actions=temp_actions)
+        self.assertEqual(type(res), type(dict()))
+        self.wait_for_service_action_status(self.vm1_name, res[self.vm1_name]['enable_vnc'])        
+
+        self.log("Enable ssh access to virtual machine (VM0)")
+        vm_vnc_url = '{}:{}'.format(self.zos_redisaddr, self.vm_vnc_port)
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/a_basic/vms_tests.py]: why not to put this on the setup .. if it is the same for all test cases?

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 1 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: done

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 2 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: done

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 3 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: done

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 4 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: done

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 5 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: done

File: tests/integration_tests/testsuite/a_basic/vms_tests.py (hunk 6 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/a_basic/vms_tests.py]: done

**Comment by Dinaamagdy** [comment]
root@dinajs9:~/zero-os/0-templates/tests/integration_tests# nosetests -v -s testsuite/a_basic/vms_tests.py  --tc-file=config.ini  
[Thu07 08:11] - ZeroOSFactoryDeprecated.py:12  :j.clients.zero_os    - WARNING  - `j.clients.zero_os` is deprecated, please use `j.clients.zos` instead
ZRT-ZOS-003 ... ok
ZRT-ZOS-004 ... SKIP: https://github.com/zero-os/0-templates/issues/140
ZRT-ZOS-005 ... ok
ZRT-ZOS-006 ... ok
ZRT-ZOS-007 ... SKIP: https://github.com/zero-os/0-templates/issues/141
ZRT-ZOS-008 [with action_type='reset'] ... ok
ZRT-ZOS-008 [with action_type='reboot'] ... ok

----------------------------------------------------------------------

**Pull request merged**